docs/conceptual/runtime/weight.md

```diff
@@ -0,0 +1,120 @@
+programming details of doing this is beyond the scope of this document. For more
+information on that, refer to:
+- the [weight
+  module's](https://crates.parity.io/sr_primitives/weights/index.html)
+  documentation in substrate-primitives.
+- the
+  [`srml-example`](https://github.com/paritytech/substrate/blob/master/srml/example/src/lib.rs)
+  module and the inline documentation. Skim the code for `#[weight]` attributes
+  and all the related documentations.
+- the section on weights in Substrate recipes. TODO
+
+The system module is responsible for accumulating the weight of each block as it
+gets executed and making sure that it does not exceed the limit. The
+transaction-payment module is responsible for interpreting these weights and
+deducting fees based upon them.
+
+
+## More on Weights: Block Weight and Length Limit
+
+Aside from affecting the fees, the main purpose of the weight system is to
+prevent a block from being too filled with transactions. The system module,
+while processing blocks≠transactions within a block, accumulates both the total
+length of the block (sum of encoded transactions in number of bytes) and the
+total weight of the blocks. At any points, if these numbers surpass the limits,
+no further transactions are accepted to the block. These limits are defined in
+[`MaximumBlockLength`](https://crates.parity.io/srml_system/trait.Trait.html#associatedtype.MaximumBlockLength)
+and
+[`MaximumBlockWeight`](https://crates.parity.io/srml_system/trait.Trait.html#associatedtype.MaximumBlockLength)
+respectively.
+
+One important note about these limits is that a ratio of them are reserved for
+the `Operational` dispatch class. This rule applies to both of the limits and
+the ratio can be found in
+[`AvailableBlockRatio`](https://crates.parity.io/srml_system/trait.Trait.html#associatedtype.AvailableBlockRatio).
+For example, if the block length limit is 1 mega bytes and the ratio is set the
+80%, all transactions can fill the first 800 kilo bytes of the block while the
+last 200 can only be filled by the operation class.
+
+
+## Frequent Questions
+
+> _Some operations get more expensive as storage grows and shrinks. How can the
+> weight system represent that?_
+
+As mentioned, weight should be known _readily_ and _ahead of dispatch_. Peeking
+the storage and analyzing it just to determine how much weight might be consumed
+does not fit this definition quite well. In this case the implementation of the
+dispatch should take the state of the change into account and manually take
+extra fees, bonds or take any other measures to make sure that the transaction
+is safe.
+
+> _Is the weighting function open to change through governance short of a
+> runtime upgrade?_
+
+Yes, it is part of a runtime itself, hence any runtime upgrade is an upgrade to
+the weight system as well. A upgrade can just update weight functions if needed.
+
+## Next Steps
+
+### Learn More
+
+TODO
+
+### Examples
+
+TODO
```

Comment:
Do we want inline code examples here? We could copy one from the recipes and then link there for more.

Reply:
I will link for sure